src/netaddress.h

```diff
@@ -39,9 +50,15 @@ enum Network
     /// IPv6
     NET_IPV6,
 
-    /// TORv2
+    /// TOR (v2 or v3)
     NET_ONION,
```

Comment:
Would it make sense to have `NET_ONION_V2` and `NET_ONION_V3`, with associated `IsOnionV2()` and `IsOnionV3()` bool helpers?

Doing so might also clean up `GetBip155Network()`, `SetIP()`, and `ToStringIP()` by removing the size checks to a better place.

It might be helpful to be able to expose network types in `getpeerinfo`, rather than trying to guess client-side like we currently have to do in #19643.

Reply:
(At some point, it might be helpful to be able to expose network types in `getpeerinfo`, rather than clients trying to guess like we currently have to do in #19643.)

Reply:
I considered suggesting this too, but outside of netaddress.*, the distinction probably doesn't matter (e.g. the decision whether to use a proxy and which one doesn't need to distinguish between TorV2 and TorV3), so this approach allows keeping the changes more local.